src/components/Views/SyncView/DeviceCell.js

```diff
@@ -0,0 +1,195 @@
+// @flow
+
+import React from 'react';
+import {
+  ActivityIndicator,
+  View,
+  Text,
+  TouchableOpacity,
+  StyleSheet,
+  Dimensions
+} from 'react-native';
+import I18n from 'react-native-i18n';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import type { Device } from '../../../types/device';
+import {
+  DARK_BLUE,
+  DARK_MANGO,
+  MANGO,
+  MAPEO_BLUE,
+  MEDIUM_BLUE,
+  VERY_LIGHT_BLUE
+} from '../../../lib/styles';
+
+type Props = {
+  device: Device,
+  onPress: (i: Device) => void,
+  selectDevice: (device: Device) => void,
+  selectedDevice?: Device,
+  updateDeviceSync: (device: Device) => void
+};
+
+const styles = StyleSheet.create({
+  cellContainer: {
+    height: 80,
+    flexDirection: 'row',
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+    borderBottomColor: MEDIUM_BLUE,
+    borderBottomWidth: 1,
+    width: Dimensions.get('window').width,
+    paddingHorizontal: 20
+  },
+  deviceName: {
+    color: 'white',
+    fontWeight: '700',
+    fontSize: 18
+  },
+  deviceTextContainer: {
+    flexDirection: 'column',
+    justifyContent: 'center',
+    alignItems: 'flex-start'
+  },
+  syncArrow: {
+    alignSelf: 'center',
+    justifyContent: 'center'
+  },
+  syncButtonContainer: {
+    backgroundColor: 'transparent',
+    position: 'absolute',
+    right: 20,
+    justifyContent: 'center',
+    alignItems: 'center'
+  },
+  syncButtonInnerCircle: {
+    width: 25,
+    height: 25,
+    borderRadius: 50,
+    backgroundColor: MANGO
+  },
+  syncButtonOuterCircle: {
+    width: 30,
+    height: 30,
+    borderRadius: 50,
+    backgroundColor: DARK_MANGO,
+    justifyContent: 'center',
+    alignItems: 'center'
+  },
+  syncStatus: {
+    color: 'white',
+    fontWeight: '400',
+    fontSize: 12
+  }
+});
+
+I18n.fallbacks = true;
+I18n.translations = {
+  en: require('../../../translations/en'),
+  es: require('../../../translations/es')
+};
+
+const DeviceCell = (props: Props) => {
+  const handlePress = () => {
+    props.onPress(props.device);
+  };
+  const handleSyncUpdate = () => {
+    const syncInProgress =
+      props.device.syncStatus === 'requested' ||
+      props.device.syncStatus === 'syncing';
+
+    if (syncInProgress) {
+      props.updateDeviceSync({
+        ...props.device,
+        syncStatus: 'stopped'
+      });
+    } else {
+      props.updateDeviceSync({
```

Comment:
hmmm you probably want to do this in a callback instead of timeouts

Reply:
oyaaa just did this to test out each stage of sync with what happens in the UI